airbyte-cdk/python/airbyte_cdk/sources/declarative/yaml_declarative_source.py

```diff
@@ -79,3 +82,74 @@ def _stream_configs(self):
             if "class_name" not in s:
                 s["class_name"] = "airbyte_cdk.sources.declarative.declarative_stream.DeclarativeStream"
         return stream_configs
+
+    @staticmethod
+    def generate_schema() -> str:
+        expanded_source_definition = YamlDeclarativeSource.expand_schema_interfaces(ConcreteDeclarativeSource, {})
+        expanded_schema = expanded_source_definition.json_schema()
+        return json.dumps(expanded_schema, cls=SchemaEncoder)
+
+    @staticmethod
+    def expand_schema_interfaces(expand_class: type, visited: dict) -> type:
+        """
+        Recursive function that takes in class type that will have its interface fields unpacked and expended and then recursively
+        attempt the same expansion on all the class' underlying fields that are declarative component. It also performs expansion
+        with respect to interfaces that are contained within generic data types.
+        :param expand_class: The declarative component class that will have its interface fields expanded
+        :param visited: cache used to store a record of already visited declarative classes that have already been seen
+        :return: The expanded declarative component
+        """
+
+        # Recursive base case to stop recursion if we have already expanded an interface in case of cyclical components
+        # like CompositeErrorHandler
+        if expand_class.__name__ in visited:
+            return visited[expand_class.__name__]
+        visited[expand_class.__name__] = expand_class
+
+        next_classes = []
+        copy_cls = type(expand_class.__name__, expand_class.__bases__, dict(expand_class.__dict__))
+        class_fields = fields(copy_cls)
+        for field in class_fields:
```

Comment:
This part of the code follows a similar structure to the original schema validation in #15543 where we unpack the underlying types and replace the annotations